### PR TITLE
Add per epoch config stress tests

### PR DESCRIPTION
**Pull request by lxfind**
## Description 

Add a test that starts two threads, one that keeps mutating the denylist, one keeps executing transactions.
Make sure there is no fork.

## Test plan 

CI

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-e2e-tests/tests/per_epoch_config_stress_tests.rs [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                        test_env
+                            .get_latest_object_ref(&test_env.regulated_coin_id)
+                            .await,
+                    )),
+                    CallArg::Pure(bcs::to_bytes(&1u64).unwrap()),
+                    CallArg::Pure(bcs::to_bytes(&DENY_ADDRESS).unwrap()),
+                ],
+            )
+            .with_type_args(vec![test_env.regulated_coin_type.clone()])
+            .build();
```
Comment (tnowacki) [inline crates/sui-e2e-tests/tests/per_epoch_config_stress_tests.rs:110]: Should we have some PTB related tests too? (i.e. transactions that do not use Move calls)

**Review by tnowacki — APPROVED** [review]

File: crates/sui-e2e-tests/tests/per_epoch_config_stress_tests.rs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+use sui_json_rpc_types::SuiTransactionBlockEffectsAPI;
+use sui_macros::sim_test;
+use sui_types::base_types::{EpochId, ObjectID, ObjectRef, SuiAddress};
+use sui_types::transaction::{CallArg, ObjectArg};
+use sui_types::{SUI_DENY_LIST_OBJECT_ID, SUI_FRAMEWORK_PACKAGE_ID};
+use test_cluster::{TestCluster, TestClusterBuilder};
+
+const DENY_ADDRESS: SuiAddress = SuiAddress::ZERO;
+
+#[ignore]
```
Comment (mystenmark) [inline crates/sui-e2e-tests/tests/per_epoch_config_stress_tests.rs:17]: why ignored?

File: crates/sui-e2e-tests/tests/per_epoch_config_stress_tests.rs [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    let gas2 = gas_objects.pop().unwrap();
+    let handle1 = {
+        let test_env = test_env.clone();
+        tokio::spawn(async move { run_transfer_thread(test_env, target_epoch, gas1.0).await })
+    };
+    let handle2 = {
+        let test_env = test_env.clone();
+        tokio::spawn(async move { run_admin_thread(test_env, target_epoch, gas2.0).await })
+    };
+    tokio::try_join!(handle1, handle2).unwrap();
```
Comment (mystenmark) [inline crates/sui-e2e-tests/tests/per_epoch_config_stress_tests.rs:38]: we might want a timeout - if reconfig halts for unrelated reasons, the test would hang forever

Comment (mystenmark) [reply crates/sui-e2e-tests/tests/per_epoch_config_stress_tests.rs:38]: also, we may want an assert that at least N transactions were executed by each thread. If i'm reading it correctly, the test could theoretically pass with only 1 transaction executed by each thread.

File: crates/sui-e2e-tests/tests/per_epoch_config_stress_tests.rs (hunk 2 above)
Comment (lxfind) [reply crates/sui-e2e-tests/tests/per_epoch_config_stress_tests.rs:17]: Because the Move APIs are not public callable yet.

**Comment by OPTIMUS-PX** [comment]
Approved

File: crates/sui-e2e-tests/tests/move_test_code/sources/regulated_coin.move [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            b"RC",
+            b"REGULATED_COIN",
+            b"A new regulated coin",
+            option::none(),
+            true,
+            ctx
+        );
+        let coin = coin::mint(&mut treasury_cap, 1000000, ctx);
+        transfer::public_transfer(coin, tx_context::sender(ctx));
+        transfer::public_transfer(deny_cap, tx_context::sender(ctx));
```
Comment (tnowacki) [inline crates/sui-e2e-tests/tests/move_test_code/sources/regulated_coin.move:22]: For extra fanciness
```suggestion
        transfer::public_transfer(coin, ctx.sender());
        transfer::public_transfer(deny_cap, ctx.sender());
```

**Review by tnowacki — COMMENTED** [review]
Anything blocking this?

**Comment by lxfind** [comment]
No, just working on addressing all the feedback

**Comment by tnowacki** [comment]
> No, just working on addressing all the feedback

LGTM, but I'll leave for @mystenmark to stamp

**Pull request merged**